tools/dm/shard-merge.md

```diff
@@ -0,0 +1,125 @@
+
+> **Note:** To merge and replicate data from the sharded tables, you must configure the `is-sharding: true` parameter in the task configuration file.
+
+### Restrictions
+
+DM has the following sharding DDL usage restrictions:
+
+- In a logical **sharding group** (composed of all sharded tables that need to be merged and replicated into one same downstream table), the same DDL statements must be executed in the same order in all upstream sharded tables (the schema name and the table name can be different).
+    - For example, if you add `column A` to `table_1` before you add `column B`, then you cannot add `column B` to `table_2` before you add `column A`. Executing the DDL statements in a different order is not supported.
+- If multiple sharding groups exist in a task, you cannot start to execute the DDL statements in other sharding groups until the DDL statements in one sharding group has been replicated successfully.
+    - For each sharding group, it is recommended to use one independent task to perform the replication.
+- In a sharding group, the corresponding DDL statements should be executed in all upstream sharded tables.
+    - For example, if DDL statements are not executed on one or more upstream sharded tables corresponding to `DM-worker-2`, then other DM-workers that have executed the DDL statements will pause their synchronization task and wait for `DM-worker-2` to receive the upstream DDL statements.
+- In a sharding group, the next DDL statement cannot be executed unless the current DDL operation is completely finished.
+- The sharding group replication task does not support `DROP DATABASE/DROP TABLE`.
+    - The Syncer unit in DM-worker automatically ignores the `DROP DATABASE/DROP TABLE` statement of upstream sharded tables.
+- The sharding group replication task supports `RENAME TABLE`, but with the following limitations:
+    - A table can only be renamed to a new name that is not used by any other table.
+    - A single `RENAME TABLE` statement can only involve a single `RENAME` operation. (Online DDL is supported in another solution)
+- The table schema of each sharded table must be the same at the starting point of the incremental replication task, so as to make sure the DML statements of different sharded tables can be replicated into the downstream with a definite table schema, and the subsequent sharding DDL statements can be correctly matched and replicated.
+- If you need to change the [table routing](../../tools/dm/data-synchronization-features.md#table-routing) rule, you have to wait for the replication of all sharding DDL statements to complete.
+    - During the replication of sharding DDL statements, an error is reported if you use `dmctl` to change `router-rules`.
```

Comment:
do we have a test case for it? @amyangfei @csuzhangxc

Reply:
I think no test case exists, and we only add a check in DM.

Reply:
we should make sure it works